### PR TITLE
Add mention to dont_merge_cookies in CookiesMiddlewares docs (#2999)

### PR DESCRIPTION
**Pull request by raphapassini**
Add mention to dont_merge_cookies in the Documentation of CookiesMIddlewares issue: #2999

### CODE REVIEW

File: docs/topics/downloader-middleware.rst [hunk 1]
```diff
@@ -190,6 +190,12 @@ The following settings can be used to configure the cookie middleware:
 * :setting:`COOKIES_ENABLED`
 * :setting:`COOKIES_DEBUG`
 
+Notice that if the :class:`~scrapy.http.Request` 
+has ``meta['dont_merge_cookies']`` present then
```
Comment (kmike) [inline docs/topics/downloader-middleware.rst]: It can be present, but False; I expect cookies to be merged in this case. Is it how it works?

Comment (kmike) [reply docs/topics/downloader-middleware.rst]: What do you think about adding more links - from here to somewhere in Request docs, and back?

File: docs/topics/downloader-middleware.rst [hunk 2]
```diff
@@ -190,6 +190,12 @@ The following settings can be used to configure the cookie middleware:
 * :setting:`COOKIES_ENABLED`
 * :setting:`COOKIES_DEBUG`
 
+Notice that if the :class:`~scrapy.http.Request` 
+has ``meta['dont_merge_cookies']`` present then 
+despite the value of :setting:`COOKIES_ENABLED` the cookies received in the
```
Comment (kmike) [inline docs/topics/downloader-middleware.rst]: This has a small readability flow issue: we haven't explained what COOKIES_ENABLED does yet. 
What do you think about moving it somewhere after COOKIES_ENABLED docs?
Alternatively, we can just mention that middleware is controlled by this key, and refer to Request docs.

**Comment by raphapassini** [comment]
Hey @kmike I did some improvements in the documentation, tells me how it looks to you.

File: docs/topics/downloader-middleware.rst [hunk 3]
```diff
@@ -226,6 +226,17 @@ Default: ``True``
 Whether to enable the cookies middleware. If disabled, no cookies will be sent
 to web servers.
 
+Notice that if the :class:`~scrapy.http.Request` 
+has ``meta['dont_merge_cookies']`` evaluates to ``True``.
```
Comment (kmike) [inline docs/topics/downloader-middleware.rst]: I'm not a native speaker, but `has ... evaluated` sounds more correct

**Pull request merged**